### ISSUE TITLE
chromium fails to regain keyboard focus after popup has been closed

### ISSUE THREAD

**Issue by spectrum70** [labwc/labwc#226]
Hi all !

adding this in the list, just to avoid to forget about it, not sure if it was already tracked, in case just mark it invalid.

Using 0c0c54fe43980a23b3064bcffe86bcd2b064c712 on Xwayland

When chromium or google-.chrome-stable drives site-related notification popups in the bottom right corner, the popup stays there, as not clickable, not possible to close it.

**Comment by Consolatis** [comment]
@spectrum70 could you test the v0.5 branch and verify the issue is fixed?

**Comment by spectrum70** [comment]
Hi,

tested, still there.

Google calendar popup comes out in the bottom right corner,
mouse have no effect over it, as if it is just an image.

**Comment by johanmalm** [comment]
Would you mind trying with the scene-graph branch

**Comment by spectrum70** [comment]
hi,

tested scene-graph with wlroot master,
labwc starts without second monitor working, and crashes
in few seconds. This branch (or wlroot master) seems to
have issues, cannot proceed in testing.

**Comment by Consolatis** [comment]
@spectrum70 are you sure its actually using xwayland instead of running native wayland? If you `A-Tab` it should state either `[xdg-shell]` or `[xwayland]` in front of the application.
If running under xwayland PR #247 should actually have fixed this.

Test run on branch v0.5:
```
$ echo $DBUS_SESSION_BUS_ADDRESS

$ chromium 'https://www.bennish.net/web-notifications.html'
[..]
$ wmctrl -l
0x00400003  0 odroidc4 HTML5 Web Notifications Test - Bennish.net - Chromium
```
```
This is the output console.
Text will appear here when stuff happens.
------------------
Permission to display: granted
Notification #1 queued for display
Notification #1 showed
Notification #1 clicked
Notification #1 clicked
Notification #1 closed
```

There is the concept of dbus notifications which may actually be the issue here:
Could you try opening a terminal and run `unset DBUS_SESSION_BUS_ADDRESS; chromium`
If that one works we may actually have an issue with input on wayland native overlay popups instead of X11 ones.

<details><summary>scene-graph</summary>

> tested scene-graph with wlroot master,
> labwc starts without second monitor working, and crashes
> in few seconds.

Might be related to the compiler mixing up two different versions, had similar issues in the past.
I am using two different build directories for that reason: `meson build_v0.15` and `meson build_master`
</details>

**Comment by Consolatis** [comment]
Current master does not contain the fixes, it is temporarily frozen until the scene-graph branch works 'good enough' at which point it will be merged into master.

The fix for X11 based popups is only available in branch v0.5 and the fixes for wayland native overlay popups is only available in my forked repo, will send a PR to get that merged in the v0.5 branch as well.

**Comment by Consolatis** [comment]
#247 and #257 should fix the missing cursor input for both cases, X11 and wayland native based notifications.

@spectrum70 Please verify branch v0.5 fixes it for you as well. You might need to create a new build and wlroots directory so it doesn't get mixed up with different versions.

Something like:
```
# clean wlroots
$ rm -rf subprojects/wlroots
# select branch and update
$ git checkout v0.5
$ git pull
# create new build directory
$ meson build_v0.5
$ meson compile -C build_v0.5
# and test
$ ./build_v0.5/labwc
```

**Comment by spectrum70** [comment]
Tested v.05, with the same above test web page, it works, but just after, keyboard is not working anymore in the google-chrome browser, i have to close and reopen it.

**Comment by 01micko** [comment]
Just as an update.. I managed to get latest `vivaldi-snapshot` running as an `xdg-shell` (aka under wayland) and the bennish tests pass. For some reason latest stable only runs as Xwayland. I pinged @ruario (former opera, curerent vivaldi dev - on twitter) about this.

**Comment by acoolstraw** [comment]
this still happens, notifications break the browser

**Comment by Consolatis** [comment]
This has very likely been fixed by 61675b521c10496f289f88b230ec209d7219d523.
If it isn't, feel free to reply here and we'll reopen the isse.
